core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpSamDAO.scala

```diff
@@ -75,11 +104,47 @@ class HttpSamDAO(baseSamServiceURL: String, serviceAccountCreds: Credential)(imp
     }
   }
 
-  override def getPolicySyncStatus(resourceTypeName: SamResourceTypeName, resourceId: String, policyName: SamResourcePolicyName, userInfo: UserInfo): Future[SamPolicySyncStatus] = {
-    val url = samServiceURL + s"/api/google/v1/resource/${resourceTypeName.value}/$resourceId/$policyName/sync"
+  private class SamApiCallback[T](functionName: String) extends ApiCallback[T] {
+    private val promise = Promise[T]()
+
+    override def onFailure(e: ApiException, statusCode: Int, responseHeaders: util.Map[String, util.List[String]]): Unit = {
+      val response = e.getResponseBody
```

Comment:
Why is L111-130 needed? We're constructing a `Future[Try[...]]` from the response body but nothing is using the result… we're just completing the promise with the original `ApiException`.

Reply:
that's an oversight, it should be completing the promise with a `RawlsExceptionWithErrorReport` so that the retry logic works the same